### PR TITLE
Server names and admin log full-text search

### PR DESCRIPTION
**Pull request by juliangiebel**
## About the PR 

This PR adds server names to rounds as a separate entity and full-text search for admin log messages on postgres.
Also fixes EF Core tools for postgress

Changes that where supposed to be in #5863 are in this PR

### CODE REVIEW

**Comment by juliangiebel** [comment]
From the PR I accidentally killed:
Fixes the Constructor for the PostgresDBContext so the (service provicer?) can actually find it.
Adds CreateDbContext from IDesignTimeDbContextFactory for tools like dotnet ef

When using dotnet ef tools you have to add a -- at the end and then add the connection string for your database.
Also you need to provide the project and context to use to dotnet ef.
Example:
dotnet-ef database update --project .\SS14\Content.Server.Database\Content.Server.Database.csproj --context PostgresServerDbContext -- "Server=127.0.0.1;Port=5432;Database=ss14;User Id=<username here>;Password=<password here>"

File: Content.Server.Database/Model.cs
Comment (PJB3005) [inline Content.Server.Database/Model.cs:283]: All these attributes you added are automatically inferred by EFCore, you can get rid of them.

File: Content.Server/Database/ServerDbBase.cs
Comment (PJB3005) [inline Content.Server/Database/ServerDbBase.cs:535]: You should use `SingleOrDefaultAsync()` here.

File: Content.Server/GameTicking/GameTicker.RoundFlow.cs
Comment (PJB3005) [inline Content.Server/GameTicking/GameTicker.RoundFlow.cs:206]: There are multiple problems with using this value:
1. This value is intended for stuff like client-side storage of rules and changelog info. It is NOT unique between our servers.
2. The default value for this (because yes, people WILL forget to set this) is different from the default fallback "unknown" value that you put in the `server` table.

Make a new one instead that we can keep internally in our infrastructure instead of being public facing like this.

**Review by PJB3005 — CHANGES_REQUESTED** [review]

File: Content.Server.Database/Model.cs
Comment (juliangiebel) [reply Content.Server.Database/Model.cs]: I know that this didn't work in ss14.admin when the Rounds didn't have the inverse property annotation so I'm very unkeen about removing that but I removed the table annotation.

File: Content.Server/GameTicking/GameTicker.RoundFlow.cs
Comment (juliangiebel) [reply Content.Server/GameTicking/GameTicker.RoundFlow.cs]: I made an new one and named it adminlogs.server_name. The default value is "unknown".

**Comment by juliangiebel** [comment]
@PJB3005 Can you resolve the conflicts? I can't merge my master branch because I hard reseted it to remote/master.

File: Content.Server.Database/ModelPostgres.cs
Comment (PJB3005) [inline Content.Server.Database/ModelPostgres.cs:21]: I didn't put this in my initial review but this is a terrible idea. If the migration is a problem for this because it has to re-index, I should run it manually before merging the PR.

How heavy is this migration gonna be for us, and are there any potential problems with a sudden spike in disk usage or DB unavailability while it is happening?

**Comment by juliangiebel** [comment]
Doing it manually is defenitly safer as running the migration could take quite a while. The migration should only affect adding new admin logs when running it manually

**Comment by PJB3005** [comment]
> The migration should only affect adding new admin logs when running it manually

Can the admin log code handle having the database be effectively write locked for the duration of the index build?

**Comment by PJB3005** [comment]
Actually I should be able to add the index online, so no concern there I guess.

**Review by PJB3005 — APPROVED** [review]

**Comment by PJB3005** [comment]
Migrations are broken, EFCore fails to generate them.

Also, move the migrations around so they aren't ordered completely wrong with respect to migrations that already exist in the database.

**Comment by juliangiebel** [comment]
@PJB3005 I fixed the creation of migrations 😄

**Comment by juliangiebel** [comment]
Should I remove the part of the migration that backfills the full text index so you can run that one manually?

**Comment by PJB3005** [comment]
> Should I remove the part of the migration that backfills the full text index so you can run that one manually?

Yes that would be appreciated.

**Comment by PJB3005** [comment]
This is done, I'll merge this once I have finished applying the migration on our database (need the index to build concurrently).

**Pull request merged**
